### PR TITLE
2.0release support HLS AES encryption

### PR DESCRIPTION
**Pull request by harlanc**

### CODE REVIEW

File: trunk/src/kernel/srs_kernel_utility.hpp [hunk 1]
```diff
@@ -140,6 +140,13 @@ extern char* srs_av_base64_encode(char* out, int out_size, const u_int8_t* in, i
 * output hex to data={0x13, 0x90, 0x56, 0xe5, 0xa0}
 */
 extern int ff_hex_to_data(u_int8_t* data, const char* p);
+/**
+ * convert data string to hex.
+ */
+extern char *ff_data_to_hex(char *buff, const uint8_t *src, int s, int lowercase);
```
[the remark this answers is not in the document]
Comment (harlanc) [reply trunk/src/kernel/srs_kernel_utility.hpp]: I saw that the function ff_hex_to_data and ff_hex_to_data  comes from the same source ?? is that right?then I need to rewrite this function under MIT license??

[the remark this answers is not in the document]
Comment (winlinvip) [reply trunk/src/kernel/srs_kernel_utility.hpp]: You should rewrite all code by yourself if it comes from FFMPEG or other GPL/LGPL projects.
Codes from MIT, BSD or Apache projects are ok.

[the remark this answers is not in the document]
Comment (harlanc) [reply trunk/src/kernel/srs_kernel_utility.hpp]: @winlinvip  that's okay

[the remark this answers is not in the document]
Comment (harlanc) [reply trunk/src/kernel/srs_kernel_utility.hpp]: @winlinvip  What the define of "rewrite",you know that the basic logic of the function must be the same.

[the remark this answers is not in the document]
Comment (winlinvip) [reply trunk/src/kernel/srs_kernel_utility.hpp]: You can read the RFC then write code by yourself. You can use FFMPEG to verify your result. We should never copy code from FFMEPG.

**Comment by winlinvip** [comment]
You should remove `ff_hex_to_data`.

@T-bagwell Could you please help to check all codes are OK?

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
I will remove it.

**Comment by winlinvip** [comment]
BTW, SRS2.0 is released, this feature will be merged to SRS3.

**Comment by winlinvip** [comment]
In SRS3, I have removed the GPL/LGPL functions, like `ff_hex_to_data` and `ff_av_base64`. I will appreciate if you file this feature to SRS3 branch.

**Comment by harlanc** [comment]
@winlinvip  I merged relevant codes from 2.0 to 3.0 and a PR is already established.

**Comment by winlinvip** [comment]
SRS3 will support this feature. #1109

**Pull request closed without merging**